gemini/rag-engine/rag_engine_evaluation.ipynb

```diff
@@ -0,0 +1,983 @@
+        "\n",
+        "\n",
+        "def count_files_in_gcs_bucket(bucket_path):\n",
+        "    \"\"\"\n",
+        "    Counts the number of files in a Google Cloud Storage bucket path,\n",
+        "    excluding directories.\n",
+        "\n",
+        "    Args:\n",
+        "      bucket_path: The path to the directory in the GCS bucket.\n",
+        "\n",
+        "    Returns:\n",
+        "      The number of files in the bucket path.\n",
+        "    \"\"\"\n",
+        "    # Add a trailing slash if it's not present\n",
+        "    if not bucket_path.endswith(\"/\"):\n",
+        "        bucket_path += \"/\"\n",
+        "\n",
+        "    process = subprocess.Popen(\n",
+        "        f\"gsutil ls {bucket_path}** | grep -v '/$' | wc -l\",\n",
+        "        shell=True,\n",
+        "        stdout=subprocess.PIPE,\n",
+        "        stderr=subprocess.PIPE,\n",
+        "    )\n",
+        "    stdout, stderr = process.communicate()\n",
+        "    if stderr:\n",
+        "        raise RuntimeError(f\"Error counting files in GCS: {stderr.decode()}\")\n",
+        "    return int(stdout.decode().strip())\n",
+        "\n",
+        "\n",
+        "def count_directories_after_split(gcs_bucket_path):\n",
+        "    \"\"\"\n",
+        "    Counts the number of directories in a Google Cloud Storage bucket path.\n",
+        "\n",
+        "    Args:\n",
+        "      gcs_bucket_path: The path to the directory in the GCS bucket.\n",
+        "\n",
+        "    Returns:\n",
+        "      The number of directories in the bucket path.\n",
+        "    \"\"\"\n",
+        "    num_files_in_bucket = count_files_in_gcs_bucket(gcs_bucket_path)\n",
+        "    num_directories = math.ceil(num_files_in_bucket / 10000)\n",
+        "    return num_directories"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Me5Ee8yCTDpo"
+      },
+      "source": [
+        "### Convert BEIR corpus to `RagCorpus` format and upload to GCS bucket."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "D0j_ceMjXddK"
+      },
+      "outputs": [],
+      "source": [
+        "CONVERTED_DATASET_PATH = f\"/converted_dataset_{dataset}\"\n",
+        "# Convert BEIR corpus to RAG format.\n",
+        "convert_beir_to_rag_corpus(corpus, CONVERTED_DATASET_PATH)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "cQd2c173rPeP"
+      },
+      "source": [
+        "#### Create a test bucket for uploading BEIR evaluation dataset to (or use an existing bucket of your choice)."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "0mG26u-QrOO2"
+      },
+      "outputs": [],
+      "source": [
+        "!gsutil mb gs://beir-test-bucket"
+      ]
+    },
```

Comment:
Change this to use a variable instead of hardcoding your bucket name

Reply:
Done.